File: src/backend/media-service/server/src/kafka/channel-event.dto.ts
import { IsNotEmpty, IsNumber, IsString } from 'class-validator';

export class ChannelEventDto {
  @IsNotEmpty()
  @IsNumber()
  guildId: number;

  @IsNotEmpty()
  @IsNumber()
  channelId: number;

  @IsNotEmpty()
  @IsNumber()
  userId: number;

  @IsNotEmpty()
  @IsString()
  type: string;
}

File: src/backend/chat-service/src/main/java/harmony/chatservice/dto/response/ChannelEventDto.java
package harmony.chatservice.dto.response;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ChannelEventDto {

    private Long guildId;
    private Long channelId;
    private Long userId;
    private String type;
}